packages/flame/lib/src/geometry/shape.dart

```diff
@@ -116,25 +131,19 @@ abstract class Shape {
   Set<Vector2> intersections(Shape other) {
     return intersection_system.intersections(this, other);
   }
-
-  /// Turns a [Shape] into a [ShapeComponent]
-  ///
-  /// Do note that while a [Shape] is defined from the center, a
-  /// [ShapeComponent] like all other components default to an [Anchor] in the
-  /// top left corner.
-  ShapeComponent toComponent({Paint? paint}) {
-    return ShapeComponent(this, paint: paint);
-  }
 }
 
 mixin HitboxShape on Shape {
   late PositionComponent component;
 
   @override
-  Vector2 get size => component.scaledSize;
+  Vector2 get size => component.size;
+
+  @override
+  Vector2 get scale => component.absoluteScale;
```

Comment:
I suspect that for non-symmetrical shapes taking absolute scale will result in wrong collision detection if the shape was flipped

Reply:
True, I have not taken flipping into consideration yet (this is still a draft).
In the future (not in this PR) I will try to refactor `Polygon` to use `Path` instead, which supports a matrix transform method.